dnscache.go

```diff
@@ -31,6 +31,11 @@ type Resolver struct {
 	OnCacheMiss func()
 }
 
+type ResolverRefreshOptions struct {
+	clearUnused      bool
+	persistOnFailure bool
```

Comment:
Those need to be public to be usable.

Reply:
Yikes, sorry about that sir. Fixed. I apologize for making the PR before using it in anger. Going to run it through some OAT on monday. Big thanks for your time.